### PR TITLE
Avoid warning in HTTP/2 code and add more logging

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala [hunk 1]
```diff
@@ -165,7 +165,8 @@ private[http2] class Http2ServerDemux(http2Settings: Http2ServerSettings, initia
                 multiplexer.pushControlFrame(SettingsAckFrame(settings))
               }
 
-            case PingFrame(true, _) =>
+            case SettingsAckFrame(Nil) => // This should be the ack to the initial settings frame
```
Comment (raboof) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala:168]: So we don't care and just drop the ack? Seems a bit suspicious but :+1:

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala [hunk 2]
```diff
@@ -67,10 +67,13 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with StageLoggi
...
   }
   private def updateState(streamId: Int, handle: IncomingStreamState => IncomingStreamState): Unit = {
-    handle(streamFor(streamId)) match {
+    val oldState = streamFor(streamId)
+    val newState = handle(oldState)
+    newState match {
       case Closed   => incomingStreams -= streamId
       case newState => incomingStreams += streamId -> newState
     }
+    log.debug(s"Incoming stream state updated for [$streamId] ${oldState.stateName} -> ${oldState.stateName}")
```
Comment (raboof) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala:76]:
```suggestion
    log.debug(s"Incoming stream state updated for [$streamId] ${oldState.stateName} -> ${newState.stateName}")
```

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala (hunk 1 above)
Comment (ennru) [reply akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala:168]: It was just logged as unhandled before (debug level).

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala (hunk 2 above)
Comment (jrudolph) [reply akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala]: Ehem, yes, thanks :)

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala (hunk 1 above)
Comment (raboof) [reply akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala:168]: Yes, in that sense the behavior doesn't really change, but handling it explicitly like this gives the impression that it is 'correct' to just ignore the Ack, which was surprising to me - you'd expect it to be in the protocol for a reason.

Comment (jrudolph) [reply akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala:168]: Yes, exactly, and it's an answer to the frame we send in `preStart`. Should we start sending proper things in there, we need to revise.

Comment (raboof) [reply akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala:168]: for now perhaps add a comment here explaining what we should ideally do (even though we don't do it now)?

Comment (jrudolph) [reply akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: Good point, added https://github.com/akka/akka-http/issues/3185 and linked to it from the comment.

**Pull request merged**